plugins/modules/ndo_l3out_template.py

```diff
@@ -0,0 +1,900 @@
+                originate_default_route=dict(type="str", choices=list(ORIGINATE_DEFAULT_ROUTE)),
+                originate_default_route_always=dict(type="bool", aliases=["always"]),
+            ),
+        ),
+        bgp=dict(
+            type="dict",
+            options=dict(
+                inbound_route_map=dict(type="str", aliases=["import_route", "inbound_route"]),
+                outbound_route_map=dict(type="str", aliases=["export_route", "outbound_route"]),
+                route_dampening_ipv4=dict(type="str", aliases=["dampening_ipv4"]),
+                route_dampening_ipv6=dict(type="str", aliases=["dampening_ipv6"]),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["vrf", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    l3out_template = module.params.get("l3out_template")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    l3_domain = module.params.get("l3_domain")
+    pim = module.params.get("pim")
+    interleak = module.params.get("interleak")
+    static_route_redistribution = module.params.get("static_route_redistribution")
+    connected_route_redistribution = module.params.get("connected_route_redistribution")
+    attached_host_route_redistribution = module.params.get("attached_host_route_redistribution")
+    target_dscp = TARGET_DSCP_MAP.get(module.params.get("target_dscp"))
+    vrf_dict = module.params.get("vrf") if module.params.get("vrf") else {}
+    ospf = module.params.get("ospf") if module.params.get("ospf") else {}
+    bgp = module.params.get("bgp") if module.params.get("bgp") else {}
+    state = module.params.get("state")
+
+    l3out_identifier = None
+    if state in ["absent", "present"]:
+        l3out_identifier = "Name: {0}".format(name) if name is not None else "UUID: {0}".format(uuid)
+
+    l3out_template_object = MSOTemplate(mso, "l3out", l3out_template)
+    l3out_template_object.validate_template("l3out")
+
+    tenant_id = l3out_template_object.template_summary.get("tenantId")
+    tenant_name = l3out_template_object.template_summary.get("tenantName")
+
+    l3outs = l3out_template_object.template.get("l3outTemplate", {}).get("l3outs", [])
+
+    if state in ["query", "absent"] and l3outs == []:
+        mso.exit_json()
+    elif state == "query" and not (name or uuid):
+        mso.existing = l3outs
+
+    l3out_object = (None, None)
+    if l3outs and (name or uuid):
+        l3out_kv_list = []
+
+        if uuid:
+            l3out_kv_list = [KVPair("uuid", uuid)]
+        else:
+            l3out_kv_list = [KVPair("name", name)]
```

Comment:
why define this empty list and then not append but overwrite list? 
```suggestion
        l3out_kv_list = [KVPair("name", name)]
        if uuid:
            l3out_kv_list = [KVPair("uuid", uuid)]
```